back/config/config.go

```diff
@@ -13,115 +14,53 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-const (
-	stagingEnv    string = "staging"
-	productionEnv string = "production"
-)
-
-type Config struct {
-	Env        Env `yaml:"env"`
-	Handler    ClientHandler
-	Production ProductionConfig
-	Staging    StagingConfig
-}
-
-type StagingConfig struct {
-	Env struct {
-		Name string `yaml:"name"`
-	} `yaml:"staging"`
-}
-
-type ProductionConfig struct {
-	Env struct {
-		Name string `yaml:"name"`
-	} `yaml:"production"`
-}
-
-type Env struct {
-	BuildProduction bool    `yaml:"build_production"`
-	Text            Text    `yaml:"text"`
-	Secrets         Secrets `yaml:"secrets"`
-}
-
-type ClientHandler struct {
-	httpClient  httpclient.HttpApi
-	VaultClient vault.VaultClient
-}
-
-type HashicorpVault struct {
-	AppName        string `yaml:"app_name"`
-	AuthURL        string `yaml:"authentication-url"`
-	BaseURL        string `yaml:"base_url"`
-	OrganizationID string
-	ProjectID      string
-}
-
-type Text struct {
-	HashicorpVault HashicorpVault `yaml:"vault"`
-}
-
-type Secrets struct {
-	StripeId      string `yaml:"STRIPE_ID"`
-	StripeToken   string `yaml:"STRIPE_TOKEN"`
-	SendgridId    string `yaml:"SENGDRID_ID"`
-	SendgridToken string `yaml:"SENDGRID_TOKEN"`
-}
+func InitializeConfig() (Config, error) {
+	var config Config
 
-func InitializeConfig() (config Config, err error) {
 	envFile, err := os.ReadFile("config/default.yaml")
 	if err != nil {
-		return Config{}, errors.Wrapf(err, "InitializeConfig, unable to read config file")
+		return Config{}, errors.Errorf("InitializeConfig, unable to read config file %s", err)
 	}
 
 	if err = yaml.Unmarshal(envFile, &config); err != nil {
-		return Config{}, errors.Wrapf(err, "InitializeConfig, unable to unmarshal")
+		return Config{}, errors.Errorf("InitializeConfig, unable to unmarshall %s", err)
```

Comment:
It's switched to an ErrorF because it's not a parent function. 

You can check more about it  (here)[https://github.com/KrispyTech/airneis/blob/staging/docs/coding-practices.md#error-handling]

Reply:
Thank you for your documentation